catalog/cataloger_diff.go

```diff
@@ -112,45 +185,130 @@ func (c *cataloger) getDiffSummary(ctx context.Context, tx db.Tx) (map[Differenc
-	// get the last child commit number of the last parent merge
-	// if there is none - then it is  the first merge
-	var maxChildMerge CommitID
-	childLineage, err := getLineage(tx, childID, UncommittedID)
-	if err != nil {
-		return fmt.Errorf("child lineage failed: %w", err)
-	}
-	parentLineage, err := getLineage(tx, parentID, CommittedID)
-	if err != nil {
-		return fmt.Errorf("parent lineage failed: %w", err)
-	}
-	maxChildQuery, args, err := sq.Select("MAX(commit_id) as max_child_commit").
+func (c *cataloger) diffFromParent(ctx context.Context, tx db.Tx, parentID, childID int64, limit int, after string) error {
+	// get child last commit of merge from parent
+	var childLastFromParentCommitID CommitID
+	query, args, err := sq.Select("MAX(commit_id) as max_child_commit").
 		From("catalog_commits").
 		Where("branch_id = ? AND merge_type = 'from_parent'", childID).
 		PlaceholderFormat(sq.Dollar).
 		ToSql()
 	if err != nil {
 		return fmt.Errorf("get child last commit sql: %w", err)
 	}
-	err = tx.Get(&maxChildMerge, maxChildQuery, args...)
+	err = tx.Get(&childLastFromParentCommitID, query, args...)
 	if err != nil {
 		return fmt.Errorf("get child last commit failed: %w", err)
 	}
-	diffResultsTableName, err := diffResultsTableNameFromContext(ctx)
+
+	diffResultsTableName, err := createDiffResultsTable(ctx, tx)
 	if err != nil {
 		return err
 	}
-	diffFromParentSQL, args, err := sqDiffFromParentV(parentID, childID, maxChildMerge, parentLineage, childLineage).
-		Prefix(`CREATE UNLOGGED TABLE ` + diffResultsTableName + " AS ").
-		PlaceholderFormat(sq.Dollar).
-		ToSql()
+
+	scannerOpts := DBScannerOptions{
+		After:            after,
+		AdditionalFields: []string{DBEntryFieldChecksum},
+	}
+	parentScanner := NewDBLineageScanner(tx, parentID, CommittedID, &scannerOpts)
+	childScanner := NewDBLineageScanner(tx, childID, UncommittedID, &scannerOpts)
+	childLineage, err := childScanner.ReadLineage()
 	if err != nil {
-		return fmt.Errorf("diff from parent sql: %w", err)
+		return err
+	}
+	batch := newDiffResultsBatchWriter(tx, diffResultsTableName)
+	var childEnt *DBScannerEntry
+	records := 0
+	for parentScanner.Next() {
+		// is parent element is relevant
+		parentEnt := parentScanner.Value()
+
+		// get next child entry - scan until we match child's path to parent (or bigger)
+		childEnt, err = ScanDBEntryUntil(childScanner, childEnt, parentEnt.Path)
+		if err != nil {
+			return fmt.Errorf("scan next child element: %w", err)
+		}
+
+		// point to matched child based on path
+		var matchedChild *DBScannerEntry
+		if childEnt != nil && childEnt.Path == parentEnt.Path {
+			matchedChild = childEnt
+		}
+		// diff between entries
+		diffType := evaluateFromParentElementDiffType(childID, childLastFromParentCommitID, childLineage, parentEnt, matchedChild)
+		if diffType == DifferenceTypeNone {
+			continue
+		}
+
+		diffRec := &diffResultRecord{
+			SourceBranch: parentID,
```

Comment:
I don't understand: isn't this ID fixed on each record, and equal to a parameter that the caller passed in?  (If so, can we just not include it?)

Reply:
right, it is a bug - should take the branch ID from the parent entry's branch